codeowners, roachtest: merge sql-schema, sql-sessions to sql-foundations

celiala opened the pull request:
Reflecting merge sql-schema, sql-sessions to sql-foundations in:

- [x] github CODEOWNERS
- [x] pkg/cmd/roachtest owners
- [x] TEAMS.yaml

Tests should pass once new team is created via https://github.com/cockroachlabs/crl-infrastructure/pull/775

This is a task for Part 2, below.
____

### Tasks to update: GitHub Team name + GitHub Projects + Blathers

Part 1
- [x] Create GitHub T- label: https://github.com/cockroachdb/cockroach/labels/T-sql-foundations
- [ ] Create new sql-foundations GitHub team name: https://github.com/cockroachlabs/crl-infrastructure/pull/775

Part 2

- [ ] Update Blathers: https://github.com/cockroachlabs/blathers-bot/pull/123
- [ ] Update CODEOWNERS, roachtest, TEAMS.yaml:
  - [ ] master: https://github.com/cockroachdb/cockroach/pull/102785
  - [ ] release-23.1: https://github.com/cockroachdb/cockroach/pull/102924 
  - [ ] release-22.2: https://github.com/cockroachdb/cockroach/pull/102925
  - [ ] release-22.1: https://github.com/cockroachdb/cockroach/pull/103536
- [ ] [Manual Step] To be done at same time as merging Part 2 PRs:
  - [ ] Manually rename "SQL Schema" Project to "SQL Foundations": https://github.com/cockroachdb/cockroach/projects/47

Part 3

- [ ] Remove old GitHub team names: https://github.com/cockroachlabs/crl-infrastructure/pull/776

____


Partial work for DEVINFHD-867
Release note: None
Epic: DEVINFHD-867

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/102785)
<!-- Reviewable:end -->

celiala commented on TEAMS.yaml:
```diff
@@ -21,14 +21,14 @@ cockroachdb/docs:
     cockroachdb/sql-syntax-prs: other
     cockroachdb/sqlproxy-prs: other
     cockroachdb/sql-api-prs: other
   triage_column_id: 7259065
```
TODO: figure out how the team merge should affect GitHub Projects.

celiala replied:
Discussed with team -- for GitHub projects, we want to:
- rename "SQL Schema" project to "SQL Foundations"
- deprecate "SQL Sessions" project

See for details: https://cockroachlabs.atlassian.net/browse/DEVINFHD-867

celiala committed:
codeowners, roachtest: merge SQL schema + sessions to SQL foundations

The SQL Schema and Sessions teams are merging to form SQL
Foundations. As described in DEVINFHD-867 we want to:

- reroute old teams to new sql-foundations team
- rename "SQL Schema" project to "SQL Foundations"
- deprecate "SQL Sessions" project

Release note: None

celiala commented:
blathers backport 23.1 22.2

celiala commented on TEAMS.yaml:
```diff
@@ -21,14 +21,7 @@ cockroachdb/docs:
     cockroachdb/docs-infra-prs: other
-cockroachdb/sql-sessions:
-  aliases:
-    cockroachdb/sql-syntax-prs: other
-    cockroachdb/sqlproxy-prs: other
-    cockroachdb/sql-api-prs: other
```
ah, I didn't think through this. you're right:
- I'll map these aliases over to the sql-foundations project (and apply to both this and the backports)

thank you for the catch!

celiala replied:
TFTR! This is RFAL:

Changes made to `TEAMS.yaml` for the renamed `cockroachdb/sql-foundations` team:
- aliases have been kept/mapped to the renamed `cockroachdb/sql-foundations` team
- label for new `T-sql-foundations` team has been mapped to the renamed `cockroachdb/sql-foundations` team

These changes have also been applied to both backports:
- for release-23.1: https://github.com/cockroachdb/cockroach/pull/102924
- for release-22.2: https://github.com/cockroachdb/cockroach/pull/102925

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/102785)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @celiala, @herkolategan, @jlinder, and @renatolabs)

___
*[`TEAMS.yaml` line 29 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/102785#-NV6SNyS8qHdAQKg8aEx:-NV6SNyS8qHdAQKg8aEy:b-wkaixc) ([raw file](https://github.com/cockroachdb/cockroach/blob/32803022c35131014f385d669816c2823b77aa5a/TEAMS.yaml#L29)):*
> ```YAML
>     cockroachdb/sqlproxy-prs: other
>     cockroachdb/sql-api-prs: other
>   triage_column_id: 8946818
> ```

i think the new triage column ID is 19467489


<!-- Sent from Reviewable.io -->

celiala reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/102785)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @jlinder, @rafiss, and @renatolabs)

___
*[`TEAMS.yaml` line 29 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/102785#-NV6SNyS8qHdAQKg8aEx:-NV6eKfB2M6wTMSyTT-p:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/32803022c35131014f385d669816c2823b77aa5a/TEAMS.yaml#L29)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i think the new triage column ID is 19467489
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

celiala commented:
TFTRs!

bors r+

celiala commented:
TFTRs!

bors r=jlinder,rafiss

celiala commented:
bors r+

celiala commented:
As discussed offline, even though we no longer have any scheduled 22.1 releases, Exalate will continue to produce errors if we don't apply changes to all affected branches. So we need to also apply this change to release-22.1:

https://github.com/cockroachdb/cockroach/pull/103536
